### PR TITLE
New: Link reports to the Elements panel in devtools

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Pre-built extension for testing:
* [link-elements-panel.zip](https://github.com/webhintio/hint/files/3060778/link-elements-panel.zip)

Opening as a draft so we can start looking at how this works. This will need rebased on #2104 once that's merged.

In short `extension-browser` is now taking a JSON snapshot of the DOM instead of serializing it to HTML (other connectors currently continue to serialize, but could migrate to this approach in the future). This snapshot assigns unique IDs to nodes during serialization. These unique IDs get attached to the `ProblemLocation` of a report so they can be looked up when we want to display the originating `Node` in the Elements panel.

I've verified the approach works in both Chrome and Firefox, though we'll probably want to discuss exactly how we want this to appear in the UI. For now I stuck in a deliberately ugly button:
image

### CODE REVIEW

**Comment by molant** [comment]
@antross looks like there are some conflicts in case you haven't seen those yet.

**Comment by antross** [comment]
@molant yep still need to rebase this on the React PR now that that's merged - will take a little bit

**Comment by antross** [comment]
Looks like `packages/utils` is failing test due to insufficient coverage. I'll add tests for the new `snapshot` helper today to bring this back up.

File: packages/extension-browser/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     ],
     "timeout": "1m"
   },
-  "bundleSize": 700000,
+  "bundleSize": 650000,
```
Comment (molant) [inline packages/extension-browser/package.json]: Reducing it even more 👏👏👏

Comment (antross) [reply packages/extension-browser/package.json]: Yeah, given the recent savings I figured it was time to lock-in the lower threshold. 😉
